app/api/github/setup/route.ts

```diff
@@ -0,0 +1,206 @@
+  branch: string,
+) => {
+  try {
+    const response = await octokit.rest.repos.getContent({
+      owner,
+      repo,
+      path,
+      ref: branch,
+    });
+
+    if (Array.isArray(response.data)) {
+      return null;
+    }
+
+    return "sha" in response.data ? response.data.sha : null;
+  } catch {
+    return null;
+  }
+};
+
+const createOrUpdateFile = async (
+  octokit: Awaited<ReturnType<typeof getOctokit>>,
+  owner: string,
+  repo: string,
+  path: string,
+  content: string,
+  message: string,
+  branch: string,
+) => {
+  try {
+    const sha = await getFileSHA(octokit, owner, repo, path, branch);
+    const params = {
+      owner,
+      repo,
+      path,
+      message,
+      content: Buffer.from(content).toString("base64"),
+      branch,
+      ...(sha ? { sha } : {}),
+    };
+
+    await octokit.rest.repos.createOrUpdateFileContents(params);
+    return { success: true };
+  } catch (error: any) {
+    if (error.status === 409) {
+      throw new Error(`Conflict while updating ${path}. Please try again.`);
+    }
+    if (error.status === 422) {
+      throw new Error(
+        `Validation failed for ${path}. Please check file content and permissions.`,
+      );
+    }
+    if (error.status === 404) {
+      throw new Error(`Path ${path} not found or insufficient permissions.`);
+    }
+    throw error;
+  }
+};
+
+export const POST = async (request: NextRequest) => {
+  const { userId } = auth();
+
+  if (!userId) {
+    return NextResponse.redirect(new URL("/login", request.url));
+  }
+
+  try {
+    const octokit = await getOctokit();
+    const body = await request.json();
+
+    const validatedData = SetupRequestSchema.parse(body) satisfies SetupRequest;
+    const { repoFullName, testPattern, triggers, defaultBranch } =
+      validatedData;
+    const [owner, repo] = repoFullName.split("/");
+
+    try {
+      await octokit.rest.repos.get({
+        owner,
+        repo,
+      });
+    } catch (error) {
+      console.error("Error accessing repository:", error);
+      return NextResponse.json(
+        {
+          error:
+            "Repository not found or not accessible. Please make sure the repository exists and you have the right permissions.",
+        },
+        { status: 404 },
+      );
+    }
+
+    await createOrUpdateRepositoryConfig({
+      owner,
+      repo,
+    });
+
+    const configContent = `import type { ShortestConfig } from "@antiwork/shortest";
+
+export default {
+  headless: false,
+  baseUrl: "http://localhost:3000",
+  testPattern: '${testPattern}',
+  ai: {
+    provider: "anthropic",
+    apiKey: process.env.ANTHROPIC_API_KEY
+  }
+} satisfies ShortestConfig;`;
+
+    const workflowContent = `name: Shortest E2E Tests
+
+on:
+${triggers.onPush ? `  push:\n    branches:\n      - ${defaultBranch}\n` : ""}
+${triggers.onPullRequest ? "  pull_request:\n" : ""}
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    env:
+      ANTHROPIC_API_KEY: \${{ secrets.ANTHROPIC_API_KEY }}
```

Comment:
This assumes the project uses this ENV variable (added some [thoughts](https://github.com/antiwork/shortest/issues/384#issuecomment-2712883594) on some initial assumptions that need to be made).

If a project uses Anthropic internall, we could assume they use `ANTHROPIC_API_KEY` env.   While Shortest can use the same ENV, the user might want to separate cost from their main API key. For that reason, Shortest supports `SHORTEST_ANTHROPIC_API_KEY`, which I think it should be the default.